integrations/aws/utils/resources.py

```diff
@@ -36,21 +38,84 @@ async def describe_single_resource(
 ) -> dict[str, str]:
     async for session in get_sessions(account_id, region):
         region = session.region_name
-        async with session.client("cloudcontrol") as cloudcontrol:
-            response = await cloudcontrol.get_resource(
-                TypeName=kind, Identifier=identifier
-            )
-            resource_description = response.get("ResourceDescription")
-            serialized = resource_description.copy()
-            serialized.update(
-                {
-                    "Properties": json.loads(resource_description.get("Properties")),
-                }
-            )
-            return serialized
+        match kind:
+            case ResourceKindsWithSpecialHandling.ACM_CERTIFICATE:
+                async with session.client("acm") as acm:
+                    response = await acm.describe_certificate(CertificateArn=identifier)
+                    resource = response.get("ResourceDescription")
+                    return fix_unserializable_date_properties(resource)
+            case ResourceKindsWithSpecialHandling.AMI_IMAGE:
+                # async with session.client("imagebuilder") as imagebuilder:
+                #     response = await imagebuilder.get_image(
+                #         ImageBuildVersionArn=f"arn:aws:ec2:{region}:{account_id}:image/{identifier}"
+                #     )
+                #     resource = response.get("Image")
+                #     return fix_unserializable_date_properties(resource)
+                logger.warning(
+                    f"Skipping AMI image {identifier} because it's not supported yet"
+                )
+                return {}
+            case ResourceKindsWithSpecialHandling.CLOUDFORMATION_STACK:
+                async with session.client("cloudformation") as cloudformation:
+                    response = await cloudformation.describe_stacks(
+                        StackName=identifier
+                    )
+                    stack = response.get("Stacks")[0]
```

Comment:
is this because it returns a list with only one value?

Reply:
Yes

Reply:
The name or the unique stack ID that's associated with the stack, which aren't always interchangeable:

- Running stacks: You can specify either the stack's name or its unique stack ID.
- Deleted stacks: You must specify the unique stack ID.